### PR TITLE
feat(loggly): support http/s bulk sending for batch logs

### PR DESCRIPTION
**Pull request by bisakhmondal**
### What this PR does / why we need it:

#6112  [Phase -2 Part 1]
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by bisakhmondal** [comment]
Sample: for batch size 3 
image

File: apisix/plugins/loggly.lua [hunk 1]
```diff
@@ -232,13 +233,71 @@ local function send_data_over_udp(message)
 end
 
 
+local function send_bulk_over_http(message, metadata, conf)
+    local endpoint = path.join(metadata.value.host, "bulk", conf.customer_token, "tag", "bulk")
+    local has_prefix = metadata.value.host:find("http")
```
Comment (spacewander) [inline apisix/plugins/loggly.lua:238]: Better to use https://github.com/apache/apisix/blob/615ee41312282ca2375abddae1d074637a901d89/apisix/core/string.lua#L46

Comment (bisakhmondal) [reply apisix/plugins/loggly.lua]: Nice. Thanks for the suggestion.
Updated.

**Review by spacewander — APPROVED** [review]

File: apisix/plugins/loggly.lua [hunk 2]
```diff
@@ -232,13 +233,71 @@ local function send_data_over_udp(message)
...
+    if not res then
+        return false, "failed to write log to loggly, " .. err
+    end
+
+    if res.status ~= 200 then
+        local body = core.json.decode(res.body)
+        if not body then
+            return false, "failed to send log to loggly, http status code: " .. res.status
+        else
+            return false, "failed to send log to loggly, " .. res.body
```
Comment (tokers) [inline apisix/plugins/loggly.lua:268]: Should we also print the status code here?

Comment (bisakhmondal) [reply apisix/plugins/loggly.lua:268]: Yep. Sounds good to me. Thanks!

**Review by tokers — APPROVED** [review]

**Pull request merged**